packages/plugin-patch/sources/commands/patchCommit.ts

```diff
@@ -43,15 +43,15 @@ export default class PatchCommitCommand extends BaseCommand {
 
     const originalPath = await patchUtils.extractPackageToDisk(locator, {cache, project});
 
-    const originalPathN = npath.fromPortablePath(originalPath);
-    const patchedPathN = npath.fromPortablePath(folderPath);
+    const originalPathN = npath.fromPortablePath(originalPath).replace(/\\/g, `/`);
+    const patchedPathN = npath.fromPortablePath(folderPath).replace(/\\/g, `/`);
 
     let {stdout} = await execUtils.execvp(`git`, [`diff`, `--no-index`, originalPathN, patchedPathN], {
       cwd: this.context.cwd,
     });
 
-    stdout = stdout.replace(new RegExp(miscUtils.escapeRegExp(originalPathN), `g`), ``);
-    stdout = stdout.replace(new RegExp(miscUtils.escapeRegExp(patchedPathN), `g`), ``);
+    stdout = stdout.replace(new RegExp(miscUtils.escapeRegExp(`${originalPathN}/`), `g`), ``);
+    stdout = stdout.replace(new RegExp(miscUtils.escapeRegExp(`${patchedPathN}/`), `g`), ``);
```

Comment:
Seems like we need the following, otherwise Unix paths will break: `a/home/foo/file.txt` will remove `/home/foo/`, turning it into `afile.txt` instead of `a/file.txt`. Correct?

```suggestion
    stdout = stdout.replace(new RegExp(miscUtils.escapeRegExp(`a/${originalPathN}/`), `g`), `a/`);
    stdout = stdout.replace(new RegExp(miscUtils.escapeRegExp(`b/${patchedPathN}/`), `g`), `b/`);
```

Reply:
Right, they start with a leading slash. Correct, we need to match it more explicitly.